### ISSUE TITLE
[bug] Adding S2 Device via code manually says it requires to start with 09

### ISSUE THREAD

**Issue by sstarcher** [zwave-js/zwavejs2mqtt#2100]
### Version
zwavejs2mqtt 6.2.0
zwavejs 8.9.1

Build/Run method

- [X] PKG

Zwavejs2Mqtt version:  6.2.0
Z-Wave JS version: 8.9.1

### Describe the bug

Attempting to use Node Inclusion via Scan QR Code, but manually entered values does not take the DSK code for my device.  It says it needs to start with 09, but my code does not start with that.  Using Home Assistant to scan the code works and it does show up under `/smart-start` in the zwavejs2mqtt UI, but the node never does provision.

### To Reproduce

1. Go to control panel
2. Click on Actions -> Manage Nodes -> Next -> Scan QR Code -> Next -> Text
3. Enter the DSK number in the QR Code Text Field
4. Field says Not valid.  Must be 52 digits long and start with "90".

### Expected behavior

The device gets added to the network

### Additional context

I'm using a Aeotec G7 controller and trying to enroll in a BE469ZP.

**Comment by robertsLando** [comment]
@AlCalzone I took the dsk code validation from yours, is this wrong?

**Comment by AlCalzone** [comment]
Are you not letting the driver validate the QR code and tell you which kind it is?

**Comment by robertsLando** [comment]
> Are you not letting the driver validate the QR code and tell you which kind it is?

Yeah but I also have some validation on the input, I took it from your code to validate the qr string

**Comment by AlCalzone** [comment]
@sstarcher Am I reading correctly that you're trying to enter the DSK string where the raw QR code string is expected? I'm not sure what HomeAssistant does differently here, but the DSK is only part of the QR code string. SmartStart needs more than just the DSK. You need to enter the full QR code contents (which do start with `90`) if you're not able to scan it from within zwavejs2mqtt.

**Comment by sstarcher** [comment]
@AlCalzone you are correct.  I verified that the QR code itself does indeed start with `90`.  Shall we just convert this ticket to zwavejs2mqtt is unable to scan a QR Code.  Home assistant was able to scan it perfectly fine and my phone itself can scan the QR code, but the UI of zwavejs2mqtt fails to scan the code.

**Comment by robertsLando** [comment]
>  Shall we just convert this ticket to zwavejs2mqtt is unable to scan a QR Code

I still don't get what's the error here, if you enter the code manually and it starts with 90 do you still get the error?

**Comment by sstarcher** [comment]
No I don’t. I thought the dsk was all that was needed

**Comment by AlCalzone** [comment]
Sounds like being unable to scan the code is the actual issue here 👍🏻

**Comment by robertsLando** [comment]
@sstarcher Could you send me the qrcode? You could email me it to daniel.sorridi@gmail.com

**Comment by sstarcher** [comment]
@robertsLando I sent you a link to it in google photos via your email.  I'll remove the link this evening.

**Comment by robertsLando** [comment]
@AlCalzone By reading the issue he says:

> Enter the DSK number in the QR Code Text Field
Field says Not valid. Must be 52 digits long and start with "90".

So the issue is not when scanning the qr but when entering it in the text field, like I said the dsk code verification it's the same zwavejs uses so I don't get where the problem is. I think @sstarcher is submitting a code it's missing something

**Comment by robertsLando** [comment]
@sstarcher just checked, I didn't received anything

**Comment by sstarcher** [comment]
@robertsLando As you mentioned I entered the incorrect code.  So I would backup and say the root cause was what caused me to want to enter the code.  The QR Code did not properly scan.

**Comment by sstarcher** [comment]
I just sent it 3 minutes ago it likely has not shown up yet. 
![Screen Shot 2021-12-29 at 10 01 51 AM](https://user-images.githubusercontent.com/3778974/147680994-f1a87110-ac46-4f5f-b741-6a948a564e51.png)

**Comment by robertsLando** [comment]
@sstarcher Sorry for the delay, I dunno why it is failing, It may be something related to the underlying library used to read the qrcode

**Comment by AlCalzone** [comment]
When I implemented QR code reading in iobroker.zwave2, I noticed that some QR codes would only be scanned if the image got upscaled by a certain amount first. Maybe you're experiencing the same here.

**Comment by mundschenk-at** [comment]
I've had a similar issue with an Aeotec ZWA008. The QR code was pretty tiny and only worked once I made a photo of it and cropped it.

**Comment by robertsLando** [comment]
Is this still an issue?

**Comment by mundschenk-at** [comment]
I can't speak for the OP, but QR code scanning directly from zwavejs2mqtt is still very finicky and does not work very well for me (iOS/iPadOS). My current workaround is to read the QR code in another app and paste the text string.

**Comment by robertsLando** [comment]
Just to know, could you try with other libraries like:

https://scanapp.org/#reader
https://nimiq.github.io/qr-scanner/demo/

If one oof them is working I will replace actual scanner with that

**Comment by AlCalzone** [comment]
From my experience, some of the QR codes are simply too small to be recognized reliably.

I was in a similar boat with iobroker.zwave2 and solved it mostly by enlarging the image by 2-4x (test this out) and then running recognition on the enlarged image. The manual variant of this workaround is making a photo of the QR code with a phone, zooming in on the image and then scanning the screen 🧙‍♂️

**Comment by tommyjlong** [comment]
I'll add my experience...  I took a picture of the QR code (including the box and nearby surrounding items).   The picture image was 3024x4032. For the Smart-Start, I then imported the picture, but the result was the QR code could not be found.  I then invert cropped the picture so that only the QR code was left in the picture with the rest being white so I still had a 3024x4032 picture.  I measured the QR code size of the image which was around 725x725 pixels.  I then imported this picture and it worked.
